### DISCUSSION TITLE
Pin 15 - /IRQ & Pin 56 - PPU /WR not connected on v1.7 PCBs?

### DISCUSSION THREAD

**Discussion by DoritoShaman** [ShawMerlin/NES-Slotmaster#1] [Q&A] [answered]
Hi, I ordered a few of these direct from the PCBWay link, and pins 15 & 56 don't appear to be connected to anything on the Cartridge board. Does anyone know if these are intentional omissions? Thanks 

![1000044549-01](https://github.com/ShawMerlin/NES-Slotmaster/assets/167120044/f710d24f-49df-4bbd-8e72-757cf8bc2f42)

**Comment by ShawMerlin** [comment]
Thank you for reaching out.

There are several pins from the 72 pin connector that are not connected.
 We are using a 64 pin ribbon cable so we couldn't pass all pins through.
These pins that were selected are only used on the expansion connector and
are not needed for normal everyday use of the NES.

With future projects being developed, it is possible that more pins from
the expansion slot will be used.  I am currently working on a SMD FPC Flat
Version.
This PCB version will pass all 72 pins from the cartridge to the NES.  It
will be easy to upgrade later to this board if needed as it's only a few
screws.

https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/58feb588-f71f-43f8-9cfa-7b61d7e483c7

Thanks,
-Merlin Shaw

**Reply by DoritoShaman** [reply]
I appreciate the fast reply! 

We can only get our Slotmasters to work by bridging these pins on the cartridge board. 

![1000044570-01~2](https://github.com/ShawMerlin/NES-Slotmaster/assets/167120044/2a29b029-be00-484c-8a41-681c1977c2f1)

**Reply by ShawMerlin** [reply]
What Game are you using when testing this?

I would suggest going over all your solder joints again.  

The 6 middle pins on top and 6 middle pins on the bottom of the 72 pin connector are not used for North America games.  

There are 2 connections that we did make in order to have Expanded Audio from Japanese games that can be used with an Everdrive.  

See these pictures for the example of those pins not being used on the cartridges.  (This cartridge PCB was removed from the case to help show the pin placements)

<img width="391" alt="image" src="https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/49c932a7-89e2-4a30-ae5d-aba521447881">

<img width="394" alt="image" src="https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/9623c551-8e81-44cd-bc85-bae019922948">

**Comment by DoritoShaman** [comment]
The games I tested with are Jack Nicklaus Golf, Double Dribble, Silent Service, Hoops, and Top Gun. All have a similar look to this when those points are not bridged: 

![IMG20240414155407](https://github.com/ShawMerlin/NES-Slotmaster/assets/167120044/27736425-82dd-4740-abc4-665aa367de27)

**Reply by ShawMerlin** [reply]
Thank you for providing the list of games and the picture.  I have ordered a few of those games so I can test on my system as well.  I will also keep this thread open in case others experience this and can share their list of games.  I tested a lot of games from my Everdrive but not any that you have mentioned.

Just to note on this.  There is a pin on the 64 pin connector that is not used.  You have that pin highlighted in purple.

**Reply by ShawMerlin** [reply]
So I did a google search on Double Dribble to look at the PCB and I have found 2 different kinds of PCBs.  One is not using the middle connectors and another PCB version of the same game is using the middle connectors.

Thank you again for bringing this to our attention.  I will reply back here when I have some more information.

<img width="598" alt="image" src="https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/3414f62e-6223-45e2-91bb-f08e34f9c0f9">

**Reply by ShawMerlin** [reply]
Although, on closer inspection.  The Double Dribble on the right is still not using those pins even though the pads are there.  Pin 15 is not connected to anything.  Let me check out the other pin you were tapping into. 
<img width="438" alt="image" src="https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/1b004990-2fb7-43e8-8a18-2d56e624e9cc">

**Reply by ShawMerlin** [reply]
Are you using a NTSC NES or a PAL NTSC?

**Reply by DoritoShaman** [reply]
So I have built two slot masters, used them on two separate NTSC systems, and they both exhibited the same behavior. I ran the iron over all the pins with some flux on both, just to make sure that I didn't have any bad connections, and confirmed each pin had continuity with a voltmeter. One system had a good set of pins before, the other did not.

I don't have a better picture of my joints other than this one while I was assembling, I can take some better pictures tomorrow.

![IMG20240415124745](https://github.com/ShawMerlin/NES-Slotmaster/assets/167120044/6e8c2e27-8e97-4354-8001-27529b9dcbdd)

**Reply by ShawMerlin** [reply]
I have confirmed that Pin15 and Pin56 are not connected on the current PCBs.  From what I have been able to find, those pins were not used for the majority of games.  This might explain why all of the NES games that I have don't have a connection for 15 or 56.

I am re-evaluating what pins were chosen to see if we can include both of these pins.  I'm glad you got the games working by connecting those pins.  I will see if the ones I have connected are EXP Port pins by mistake.

Thank you again for all the info and reaching out.  I want this solution to work with all NTSC Games, not just the ones that don't use those pins.  I should have more information on this tomorrow.

**Reply by ShawMerlin** [reply, marked as answer]
DoritoShaman,

First of, I am really impressed that you were able to match those vias together without the schematic.  Great job!!

I have gone through and renumbered which pins we want to include so that we can have Pin 15 & 56.  I found that I had mapped Pin 18 and 20 which are just expansion ports that should not have use, at least not right now.  I have swapped them now so that we can still have Expansion Audio pass through.

Here are 2 quick fixes that can be made.  One is a simple fix but will not have Expansion Audio.  The second is a lot more work but will include the Expansion Audio.

**Simple Fix - This gets pin 21 back to where it should be and corrects 15 and 56.**
- Cut the Trace going into Pin18 of the 72 Pin Connector.
- Connect Pin15 to the spot that Pin18 was connected to on the 64 Pin Connector.
- Connect Pin56 to Pin32 on the 64 Pin connector (this one does not have any traces going to it)
<img width="730" alt="image" src="https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/91cf3f05-5a79-45e9-9014-da0f0d99d168">

`!`                      
**Complex Fix that includes Expansion Audio**
- 3 Cuts, removing Pin18 & Pin20 and the current Pin51 route
- Use a wire between Pin51 and the spot that Pin20 use to hook to on the 64 pin connector. (Green)
- Use a wire between Pin54 and the spot that Pin51 use to hook to on the 64 pin connector. (Green)
- Use a wire between PIn15 and the spot that Pin18 use to hook to on the 64 pin connector. (Blue)
- Use a wire between Pin56 and the Pin32 of the 64 pin connector that had no trace connected (Blue)
- On the Bottom Board, Use a wire between Pin51 of the 72 Pin Connector and Pin34 of the 64 Pin Connector (Pink)
         Both of those pins are not being used currently.

Top PCB
<img width="918" alt="image" src="https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/eb8995b0-218f-4add-82bc-23f92a2571cb">

Bottom PCB
<img width="931" alt="image" src="https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/b1648fc1-a7cc-4311-88ed-80ce66ae65f7">


Here are the Expansion Pins that were supposed to be not be passed on with the 64 pin cable.
Please note that a full 72 pin passthrough with Flat FPC Cables is in the works with the prototype ordered.
Pin 16 - EXP0
Pin 17 - EXP1
Pin 18 - EXP2
Pin 19 - EXP3
Pin 20 - EXP4
Pin 52 - EXP8
Pin 53 - EXP7
Pin 55 - EXP5

**Reply by DoritoShaman** [reply]
That's awesome! I'll get to work on the next one and let you know how it goes. Thank you for all the hard work you've done on this project!

**Comment by ShawMerlin** [comment]
I just wanted to give an update.

Today I got Double Dribble and Secret Service in the mail and confirmed as well with what you are seeing.
I will do the bodge fix tonight to also confirm your findings and then release pictures on this fix.

I also got the new V1.8 Boards today so I will be able to verify these directly against these games to make sure we are 100% compatible with all games.

Thank you so much again for reaching out so very quickly after this discovery.. This allowed the Gerbers to be taken off the repository and this new version launched and tested so quickly!!

-Merlin Shaw

**Reply by DoritoShaman** [reply]
That's awesome, glad to hear! Thanks again for developing these, they work crazy well for the price, and so fun to build! I'm happy to help in any small way I can.

**Comment by ShawMerlin** [comment]
The bodges above worked perfect so I have created this quick guide on performing this on any V1.7 Boards out in the wild.
https://github.com/ShawMerlin/NES-Slotmaster/blob/main/Board%20V1.7%20100%25%20Compatibility%20Fix.md

Thanks again for the help.
-Merlin Shaw
